lib/consensus/isaac.go

```diff
@@ -113,31 +122,90 @@ func (is *ISAAC) SelectProposer(blockHeight uint64, roundNumber uint64) string {
 	return is.proposerSelector.Select(blockHeight, roundNumber)
 }
 
-func (is *ISAAC) IsAvailableRound(round round.Round) bool {
-	// check current round is from InitRound
-	if is.LatestRound.BlockHash == "" {
+func (is *ISAAC) SaveNodeHeight(senderAddr string, height uint64) {
+	is.nodesHeight[senderAddr] = height
+}
+
+func (is *ISAAC) IsAvailableRound(round round.Round, latestBlock block.Block) bool {
+	if round.BlockHeight == latestBlock.Height {
+		if is.isInitRound(round) {
+			return true
+		}
+
+		if round.BlockHash != latestBlock.Hash {
+			return false
+		}
+
+		if round.BlockHeight == is.LatestRound.BlockHeight {
+			if round.Number <= is.LatestRound.Number {
+				return false
+			}
+		}
 		return true
 	}
 
-	if round.BlockHeight < is.latestBlock.Height {
-		return false
-	} else if round.BlockHeight == is.latestBlock.Height {
-		if round.BlockHash != is.latestBlock.Hash {
-			return false
+	return false
+}
+
+func (is *ISAAC) isInitRound(round round.Round) bool {
```

Comment:
You mean that remove this method and use like this, right?
```
if is.LatestRound.BlockHash == "" && round.BlockHeight == common.GenesisBlockHeight {
	return true
}
```
But I've added this method on purpose. Because `isInitRound` is easy to understand what the meaning is, than `is.LatestRound.BlockHash == "" && round.BlockHeight == common.GenesisBlockHeight`

Reply:
Fair enough